src/main/scala/is/hail/io/bgen/BgenRDDPartitions.scala

```diff
@@ -458,19 +431,17 @@ object CompileDecoder {
             nBitsPerProb := reader.invoke[Int]("read"),
             (nBitsPerProb < 1 || nBitsPerProb > 32).mux(
               Code._fatal(
-                const("Value for nBits must be between 1 and 32 inclusive. Found ")
+                const("nBits value must be between 1 and 32 inclusive. Found ")
                   .concat(nBitsPerProb.toS)
                   .concat(".")),
               Code._empty),
             (nBitsPerProb.cne(8)).mux(
               Code._fatal(
-                const("Only 8-bit probabilities supported, found ")
+                const("Hail only supports 8-bit probabilities, found ")
                   .concat(nBitsPerProb.toS)
                   .concat(".")),
               Code._empty),
-            nGenotypes := Code.invokeScalaObject[Int, Int](
-              BgenRDD.getClass, "triangle", nAlleles),
-            nExpectedBytesProbs := (const(settings.nSamples) * (nGenotypes - 1) * nBitsPerProb + 7) / 8,
+            nExpectedBytesProbs := (settings.nSamples * (3 - 1) * 8 + 7) / 8,
```

Comment:
of course your hardcoded version reduces to `2 * settings.nSamples` (two bytes per genotype), so if you're hardcoding anyway, you might as well just write that.  If we want to generalize down the line, it will be easy enough to bring back the triangle function and this formula, etc. (the `+ 7` is just to round up)